web/components/liquidations.tsx

```diff
@@ -75,23 +76,43 @@ export const Liquidations: React.FC = () => {
   };
 
   const redeemStx = async () => {
-    await doContractCall({
-      network,
-      contractAddress,
-      stxAddress,
-      contractName: 'arkadiko-freddie-v1-1',
-      functionName: 'redeem-stx',
-      functionArgs: [uintCV(state.balance['xstx'])],
-      postConditionMode: 0x01,
-      onFinish: data => {
-        setState(prevState => ({
-          ...prevState,
-          currentTxId: data.txId,
-          currentTxStatus: 'pending',
-        }));
-      },
-      anchorMode: AnchorMode.Any,
-    });
+    if (redeemableStx > 0) {
```

Comment:
Suggestion: contract function to encapsulate both redeem-stx methods so user only needs one transaction to redeem.